### DISCUSSION TITLE
Force client keepAliveTimeout

### DISCUSSION THREAD

**Discussion by Gianluca-Casagrande-Stiga** [moscajs/aedes#865] [Q&A] [answered]
Hi, is it possibile to force a client keep alive timeout?

Some clients don't set correctly their keep alive timeout, so their client instance remains into the broker memory even if the device is switched off for a long time.
I would like to force a 5mins keep alive timeout to every clients that doesn't have it or have it longer; if it is not possible, I would like to block them during connection authorisation.

thanks

**Comment by robertsLando** [comment, marked as answer]
Actually there is no support for this, anyway you can access to clients instances from `broker.clients` that is an object where key is client id and value is the client instance, when a client connects we set a [keepAliveTimeout](https://github.com/moscajs/aedes/blob/main/lib/handlers/connect.js#L146) that is rescheduled every time a packet is received from client: https://github.com/moscajs/aedes/blob/main/lib/handlers/index.js#L65C1-L65C1

The keep alive interval is set from the connect packet `keepAlive` timeout as shown in the first link.

If you think that this is a bug in aedes could you consider submitting a PR to fix it? This could cause memory leaks if client instances are not removed
